First PydanticAI Notebook

ofir-ov opened the pull request:
Added a demonstration (first of many) of using PydanticAI to create a GenAI agent.

This is basically a translation of the capabilities in the first notebook `simple_conversational_agent.ipynb`, but with PydanticAI. The idea is to provide more and more translations for existing notebooks so that we'll be able to show how we can use agents with PydanticAI.

If this PR is accepted, then I can move on to the next notebook (and the one after that, etc.)

NirDiamant commented:
Well written and very organized. Thanks!  

Since this is the first in the series and also appears at the beginning of the repository as a basic tutorial, could you please add a short explanation at the beginning about Pydantic agents and how they differ from other options?

ofir-ov commented:
> Well written and very organized. Thanks!
> 
> Since this is the first in the series and also appears at the beginning of the repository as a basic tutorial, could you please add a short explanation at the beginning about Pydantic agents and how they differ from other options?

Definitely! Thanks for the feedback and the suggestion.

I've added a section with a short summary of the key features and benefits of PydanticAI. I've added it at the beginning of the new notebook. LMK if you'd like me to add it some other place or make further changes 🙏

NirDiamant commented:
amazing! merged
